packages/radio/src/single-selection-controller.ts

```diff
@@ -213,15 +211,14 @@ export class SingleSelectionController {
    * @param element Element from which selection set is derived and subsequently
    *     focused.
    */
-  focus(element: CheckableElement) {
+  async focus(element: CheckableElement) {
     // Only manage focus state when using keyboard
     if (this.mouseIsDown) {
       return;
     }
     const set = this.getSet(element.name);
-    const currentFocusedSet = this.focusedSet;
```

Comment:
Why was this removed?

Reply:
It is no longer needed, currently, there is a bug that if you focus an element of the set for the second time it will not move the focus to the selected item because the set that is being focused is the `focusedSet` (it is not cleared on `blur` of the set). This was probably added for when you change the value with your keyboard as the `focus` event comes before the `update` call. By awaiting a promise we check the selected value after the `update` call so we don't have to handle the value change of keyboard anymore (the focused element will be the selected element) and a second focus will also work now.

Should probably add a check to see if the element is still focused.

There are other ways to handle this, like the `blur` handler in my comment above. That will keep the `focusedSet` and will clear the `focusedSet` when you `blur` the entire set or on mouse override.

Reply:
I can't seem to repro your issue. Is there an issue filed or can you send me a repro case?

Reply:
You can repo it here: <https://material-components.github.io/material-components-web-components/demos/formfield/>

As you can see in the GIF, the first time the first radio gets focus it is correctly setting the focus on the selected (second) radio element, but after moving the focus back to the switch and then again to the first radio it is not set to the selected but keeps the focus on the first radio element.
![radio-focus](https://user-images.githubusercontent.com/5662298/88090723-46b0ae00-cb8e-11ea-997e-9463c7078837.gif)

Reply:
I changed it to the blur logic to not mess with setting focus, that should stay sync.